src/katgpucbf/fgpu/engine.py

```diff
@@ -948,18 +948,26 @@ async def run_transmit(self) -> None:
                 await async_wait_for_events([download_marker])
 
             if dig_total_power is not None:
-                for pol, trg in enumerate(dig_total_power):
-                    total_power = float(trg)
-                    avg_power = total_power / (out_item.n_spectra * self.output.spectra_samples)
-                    # Normalise relative to full scale. The factor of 2 is because we
-                    # want 1.0 to correspond to a sine wave rather than a square wave.
-                    avg_power /= ((1 << (self.engine.src_layout.sample_bits - 1)) - 1) ** 2 / 2
-                    # If for some reason there's zero power, avoid reporting
-                    # -inf dB by assigning the most negative representable value
-                    avg_power_db = 10 * math.log10(avg_power) if avg_power else np.finfo(np.float64).min
-                    self.engine.sensors[f"input{pol}.dig-rms-dbfs"].set_value(
-                        avg_power_db, timestamp=self.engine.time_converter.adc_to_unix(out_item.end_timestamp)
-                    )
+                if np.all(out_item.present):
```

Comment:
You're making this decision on the whole item, which means that if a packet goes missing on one pol, you're setting both pols to failure. It looks like `out_item.present` has a pol axis, so you could make this more fine-grained.